text/0000-overlapping_match.md

```diff
@@ -0,0 +1,150 @@
+- Feature Name: overlapping_match_statements
+- Start Date: 2017-09-08
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This feature is to facilitate the writing and using of `match` statements in a whole new way.
+This feature would allow the ability to write match statements where multiple branches may be
+matched and still allow for code to be used if no branch is matched, similar to the current
+use of the `_` pattern.
+
+# Motivation
+[motivation]: #motivation
+
+There is a very good software engineering principle where repeating a piece of code is bad.
+This is the case because if that selection of code needs to be changed then it has to be
+changed in two places which can easily not be done and thus create bugs. A way of doing this
+for a large selection of lines of code is to put it into a function, a helper function. Allowing
+overlapping match statements extends this paradigm to that where matching is a good idea, the
+use of pattern matching, and where exhaustiveness checks are a nice thing.
+
+This would support use cases where the required execution of several branches overlapped enough
+that his would help. A use case for this is when the outcome of one branch is the same as a
+combination of the other two branches of a match statement. The expected outcome of this is
+the ability to have multiple branches of a match statement, and having those branches still be
+checked for exhaustiveness, be executed if more than one of them match the value.
+
+# Detailed design
+[design]: #detailed-design
+
+Basic Syntax:
+```rust
+match val in {
+    pat | pat => expr,
+    pat => expr
+}
+
+match val in {
+    pat | pat => expr,
+    pat => expr
+} else {
```

Comment:
This deviation from the `_ => ...` syntax used by "regular" match is a bit unfortunate

Reply:
It may be a bit unfortunate but if you consider the meaning of `_` it makes sense. The `_` pattern means to match everything.